### PR TITLE
Distributions reparametrization

### PR DESCRIPTION
**Pull request by jschueller**
This PR introduces distribution parametrization changes, among which 6 distribution breaking changes:

- New argument ordering in Frechet ctor: scale(beta), shape(alpha), location(gamma) (swapped arg1 and arg2)
- New parametrization in Gumbel: scale(beta), location(gamma) (arg1=1/arg1)
- New parametrization in Beta: shape(alpha), shape(beta), location(a), location(b) (arg2=arg1-arg2)
- New parametrization in InverseGamma: rate(lambda), shape(k) (swapped arg1 and arg2)
- New parametrization in InverseNormal: location(mu), rate(lambda) (swapped arguments))
- New parametrization in Laplace: mean(mu), rate(lambda) (swapped arguments)
The other changes in parametrization do not break scripts (parameters renaming)

Added GumbelLambdaGamma, deprecated GumbelAB.
Added LogNormalMuErrorFactor parametrization.

Also Weibull becomes WeibullMin, added WeibullMax/WeibullMaxFactory/WeibullMaxMuSigma.

Using WeibullMax in GEV instead of the mixture fixes some bugs.
Added GEVFactory (bic selection on the 3 estimations on the different models).

Now, If the user wants to maintain compatibilty with the old scripts he/she can use:
```
import openturns.shims as ot
beta = ot.Beta(r, t, a, b)
```

### CODE REVIEW

File: python/src/LogNormalMuErrorFactor_doc.i.in [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+gamma : float, optional
+    Location parameter.
+
+Notes
+-----
+The :math:`EF` parameter is defined as:
+
+.. math::
+
+    EF = \sqrt{\frac{q_{95} - \gamma}{q_{5} - \gamma}} = \exp{\sigma_{l} q_{95}^N}
```
Comment (sofianehaddad) [inline python/src/LogNormalMuErrorFactor_doc.i.in]: I suppose `q_{95}^N}`referes to centered & reduced gaussian quantile, wheras `q_{95}`refers to `log-normal`one.
Am I wrong ? Is it possible to add more details ? thanks

File: python/src/WeibullMin.i [hunk 2]
```diff
@@ -12,6 +12,6 @@ namespace OT { %extend WeibullMin { WeibullMin(const WeibullMin & other) { retur
 %pythoncode %{
 def Weibull(alpha=1.0, beta=1.0, gamma=0.0):
```
Comment (sofianehaddad) [inline python/src/WeibullMin.i]: I think it is better using `Weibull(*args*)`and check the argument

File: lib/src/Uncertainty/Distribution/WeibullMinFactory.cxx [hunk 3]
```diff
@@ -89,6 +90,17 @@ WeibullMin WeibullMinFactory::buildAsWeibullMin(const Sample & sample) const
...
 
+WeibullMin WeibullMinFactory::buildMethodOfLikelihoodMaximization(const Sample & sample) const
+{
+  const MaximumLikelihoodFactory factory(buildMethodOfMoments(sample));
+  return buildAsWeibullMin(factory.build(sample).getParameter());
+}
+
+WeibullMin WeibullMinFactory::buildAsWeibullMin(const Sample & sample) const
+{
+  return buildMethodOfLikelihoodMaximization(sample);
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Distribution/WeibullMinFactory.cxx]: Please add a Log info about the used method

File: python/src/shims.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+def Gumbel(alpha=1.0, gamma=0.0):
+    return ot.ParametrizedDistribution(ot.GumbelLambdaGamma(alpha, gamma))
+
+def Beta(r=2.0, t=4.0, a=-1.0, b=1.0):
+    return ot.Beta(r, t - r, a, b)
+
+def InverseGamma(lambda_=1.0, k=1.0):
+    return ot.InverseGamma(k, lambda_)
+
```
Comment (sofianehaddad) [inline python/src/shims.py]: Maybe the `Weibull`alias should be moved here (and dropped latter) ?

File: lib/src/Uncertainty/Distribution/GumbelAB.cxx [hunk 5]
```diff
@@ -35,14 +35,15 @@ GumbelAB::GumbelAB()
   , a_(0.0)
   , b_(1.0)
 {
-  // Nothing to do
+  LOGWARN("GumbelAB is deprecated");
 }
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/GumbelAB.cxx]: I'll remove it next release

File: lib/src/Uncertainty/Distribution/Frechet.cxx [hunk 6]
```diff
@@ -353,17 +353,17 @@ Scalar Frechet::getGamma() const
...
 }
 
 /* Method load() reloads the object from the StorageManager */
 void Frechet::load(Advocate & adv)
 {
   ContinuousDistribution::load(adv);
-  adv.loadAttribute( "alpha_", alpha_ );
   adv.loadAttribute( "beta_", beta_ );
+  adv.loadAttribute( "alpha_", alpha_ );
   adv.loadAttribute( "gamma_", gamma_ );
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/Frechet.cxx]: No problem here as the arguments are just swapped, the attributes and their meaning dont change.

File: lib/src/Uncertainty/Distribution/BetaFactory.cxx [hunk 7]
```diff
@@ -73,15 +73,15 @@ Beta BetaFactory::buildAsBeta(const Sample & sample) const
...
   {
     delta = std::max(std::abs(xMin), 100.0) * SpecFunc::ScalarEpsilon;
-    Beta result(1.0, 2.0, xMin - delta, xMax + delta);
+    Beta result(1.0, 1.0, xMin - delta, xMax + delta);
     result.setDescription(sample.getDescription());
     return result;
   }
   const Scalar mean = sample.computeMean()[0];
   const Scalar sigma = sample.computeStandardDeviationPerComponent()[0];
   const Scalar t = (b - mean) * (mean - a) / (sigma * sigma) - 1.0;
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/BetaFactory.cxx]: yes, ok

File: lib/src/Uncertainty/Distribution/Gumbel.cxx [hunk 8]
```diff
@@ -315,20 +318,49 @@ Scalar Gumbel::getBeta() const
...
+  if (adv.hasAttribute("alpha_"))
+  {
+    // old parametrization
+    Scalar alpha = 0.0;
+    Scalar beta = 0.0;
+    adv.loadAttribute( "alpha_", alpha );
+    adv.loadAttribute( "beta_", beta );
+    beta_ = 1.0 / alpha;
+    gamma_ = beta;
+  }
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/Gumbel.cxx]: ok

File: lib/src/Uncertainty/Distribution/Beta.cxx [hunk 9]
```diff
@@ -438,8 +446,21 @@ void Beta::save(Advocate & adv) const
...
+  {
+    // old parametrization
+    Scalar r = 0.0;
+    Scalar t = 0.0;
+    adv.loadAttribute( "r_", r );
+    adv.loadAttribute( "t_", t );
+    alpha_ = r;
+    beta_ = t - r;
+  }
+  else
```
[the remark this answers is not in the document]
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/Beta.cxx]: ok

File: python/src/LogNormalMuErrorFactor_doc.i.in (hunk 1 above)
Comment (jschueller) [reply python/src/LogNormalMuErrorFactor_doc.i.in]: done

File: python/src/WeibullMin.i (hunk 2 above)
Comment (jschueller) [reply python/src/WeibullMin.i]: ok, way simpler

File: lib/src/Uncertainty/Distribution/WeibullMinFactory.cxx (hunk 3 above)
Comment (jschueller) [reply lib/src/Uncertainty/Distribution/WeibullMinFactory.cxx]: ok

File: python/src/shims.py (hunk 4 above)
Comment (jschueller) [reply python/src/shims.py]: indeed

**Comment by jschueller** [comment]
@sofianehaddad what did you have in mind for compatibility ?

File: python/src/shims.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+"""
+openturns<=1.13 compatibility shims
+"""
+
+import openturns as ot
+from openturns import *
```
Comment (sofianehaddad) [inline python/src/shims.py]: why the second import ?

Comment (jschueller) [reply python/src/shims.py]: This one is to import all symbols from.

We first import all symbols, then overload our distribution ctors.

So at the end we are able to use all symbols from openturns.shims and not only our ctors:


```
import openturns.shims as ot
dist1 = ot.Arcsine()
dist2 = ot.Beta(r,t,...)
```

The only change in user scrips is importing "openturns.shims", and not:
```
import openturns as ot
import openturns.shims
dist1 = ot.Arcsine()
dist2 = openturns.shims.Beta(r, t, a, b)
```

Comment (sofianehaddad) [reply python/src/shims.py]: Ahhh I start to see what you have in mind

Comment (sofianehaddad) [reply python/src/shims.py]: `dist2` docstring refers to the new implementation docstring. Isn't it ? How to provide the correct information ? 
Btw, warning messages should be added for each of the distributions

Comment (jschueller) [reply python/src/shims.py]: That's not possible. I'm afraid.
No need to add warnings, the user used openturns.shims on purpose.

**Comment by jschueller** [comment]
@sofianehaddad do you know when you'll be able to review this (roughly) ?

**Comment by regislebrun** [comment]
+1 good job ;-)!

**Pull request merged**